src/raftstore/store/peer.rs

```diff
@@ -1391,6 +1466,9 @@ impl Peer {
         req: RaftCmdRequest,
         metrics: &mut RaftProposeMetrics,
     ) -> Result<u64> {
+        if self.is_merging {
+            return Err(box_err!("peer in read only mode, can't do proposal."));
```

Comment:
How about write this check in `get_handle_policy`? So we can deal with `is_merging` in one function.

Reply:
Because `get_handle_policy` is used to get the policy for some command instead of checking whether it's appropriate to handle a request. Besides, `propose_normal` may be called without `get_handle_policy`.